src/edu/csus/ecs/pc2/shadow/RemoteRunSubmitter.java

```diff
@@ -42,32 +42,34 @@ public RemoteRunSubmitter(IInternalController controller) {
 
     /**
      * Submit a run to the server.
-     * 
+     *
      * @param clientIdString - "teamN" client name
      * @param problemID - problem id
      * @param languageID - CLICS language id
      * @param mainFile - main file name
-     * @param auxFiles - other submittted files 
+     * @param auxFiles - other submittted files
      * @param overrideTimeMS - override submission time in MS
      * @param overrideRunId - override run id
+     * @throws Exception if run can't be submitted, eg. throttling
      */
-    public void submitRun(String clientIdString, String problemID, String languageID, IFile mainFile, List<IFile> auxFiles, long overrideTimeMS, long overrideRunId) {
+    public void submitRun(String clientIdString, String problemID, String languageID, IFile mainFile, List<IFile> auxFiles, long overrideTimeMS, long overrideRunId) throws Exception {
         submitRun(clientIdString, problemID, languageID, null, mainFile, auxFiles, overrideTimeMS, overrideRunId);
     }
-    
+
     /**
      * Submit a run to the server.
-     * 
+     *
      * @param clientIdString - "teamN" client name
      * @param problemID - problem id
      * @param languageID - CLICS language id
      * @param entry_point - entry point, null if default or none
      * @param mainFile - main file name
-     * @param auxFiles - other submittted files 
+     * @param auxFiles - other submittted files
      * @param overrideTimeMS - override submission time in MS
      * @param overrideRunId - override run id
+     * @throws Exception if run can't be submitted, eg. throttling
      */
-    public void submitRun(String clientIdString, String problemID, String languageID, String entry_point, IFile mainFile, List<IFile> auxFiles, long overrideTimeMS, long overrideRunId) {
+    public void submitRun(String clientIdString, String problemID, String languageID, String entry_point, IFile mainFile, List<IFile> auxFiles, long overrideTimeMS, long overrideRunId) throws Exception {
```

Comment:
Should throw `SubmissionRejectedException`?

Reply:
I just did what eclipse suggested.  I guess if an other exception occurs the submissions was not accepted anyway.